client/fingerprint/network_unix.go

```diff
@@ -33,41 +32,34 @@ func NewNetworkFingerprinter(logger *log.Logger) Fingerprint {
 func (f *NetworkFingerprint) Fingerprint(cfg *config.Config, node *structs.Node) (bool, error) {
 	// newNetwork is populated and addded to the Nodes resources
 	newNetwork := &structs.NetworkResource{}
-	defaultDevice := ""
-	ip := ""
-
-	// 1. Use user-defined network device
-	// 2. Use first interface found in the system for non-dev mode. (dev mode uses lo by default.)
-	if cfg.NetworkInterface != "" {
-		defaultDevice = cfg.NetworkInterface
-		ip = f.ipAddress(defaultDevice)
-	} else {
+	var ip string
 
-		intfs, err := net.Interfaces()
-		if err != nil {
-			return false, err
-		}
+	interfaces, err := f.findInterfaces(cfg.NetworkInterface)
+	if err != nil {
+		f.logger.Println(fmt.Sprintf("[DEBUG] Error while detecting network interface during fingerprinting: %s", err.Error()))
+		return false, err
+	}
 
-		for _, i := range intfs {
-			if (i.Flags&net.FlagUp != 0) && (i.Flags&(net.FlagLoopback|net.FlagPointToPoint) == 0) {
-				if ip = f.ipAddress(i.Name); ip != "" {
-					defaultDevice = i.Name
-					break
-				}
-			}
-		}
+	if len(interfaces) == 0 {
+		f.logger.Println("[DEBUG] No network interfaces were detected")
+		return false, errors.New("Unable to find any interface")
```

Comment:
I think Nomad will bail out on startup with this error so let's make it more clear. We can use the same phrasing as above:

> No network interfaces were detected